pkg/cmd/app/translate.go

```diff
@@ -0,0 +1,216 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	nameField       = "name"
+	statusField     = "status"
+	outputField     = "output"
+	repoField       = "repository"
+	branchField     = "branch"
+	filenameField   = "filename"
+	yamlField       = "yaml"
+	iconField       = "icon"
+	actionLockField = "actionLock"
+	actionNameField = "actionName"
+
+	actionDefaultName = "cli"
+
+	// ProgressingStatus indicates that an app is being deployed
+	ProgressingStatus = "progressing"
+	// DeployedStatus indicates that an app is deployed
+	DeployedStatus = "deployed"
+	// ErrorStatus indicates that an app has errors
+	ErrorStatus = "error"
+	// DestroyingStatus indicates that an app is being destroyed
+	DestroyingStatus = "destroying"
+
+	// maxLogOutput is the maximum size that we allow to allocate for logs.
+	// Specifically 800kb. The limit on configmaps is 1Mb but we want to leave some
+	// room for the other data stored in there.
+	// Note that the is the limit after encoding the logs to base64 which is how
+	// the logs are stored in the configmap.
+	maxLogOutput = 800 << (10 * 1)
+)
+
+// maxLogOutputRaw is the maximum size we allow to allocate for logs before
+// being encoded to base64
+// See: https://stackoverflow.com/a/4715480/1100238
+var maxLogOutputRaw = int(math.Floor(float64(maxLogOutput)*3) / 4)
+
+//CfgData represents the data to be include in a configmap
+type CfgData struct {
+	Name       string
+	Namespace  string
+	Status     string
+	Output     string
+	Repository string
+	Branch     string
+	Filename   string
+	Manifest   []byte
+	Icon       string
+}
+
+// TranslateConfigMap translates the app into a configMap
+func TranslateConfigMap(ctx context.Context, data *CfgData, c kubernetes.Interface) (*apiv1.ConfigMap, error) {
+	cmap, err := configmaps.Get(ctx, TranslateAppName(data.Name), data.Namespace, c)
+	if err != nil {
+		if !oktetoErrors.IsNotFound(err) {
+			return nil, err
+		}
+		cmap = translatedConfigMapSandBox(data)
+	} else {
+		if err := updateCmap(cmap, data); err != nil {
+			return nil, err
+		}
+	}
+	if err := configmaps.Deploy(ctx, cmap, cmap.Namespace, c); err != nil {
+		return nil, err
+	}
+	return cmap, nil
+}
+
+// SetOutput sets the output of a config map
+func SetOutput(cfg *apiv1.ConfigMap, output string) *apiv1.ConfigMap {
+	cfg.Data[outputField] = base64.StdEncoding.EncodeToString([]byte(output))
+	return cfg
+}
+
+// UpdateConfigMap updates the configmaps fields
+func UpdateConfigMap(ctx context.Context, cmap *apiv1.ConfigMap, data *CfgData, c kubernetes.Interface) error {
+	cmap, err := configmaps.Get(ctx, cmap.Name, cmap.Namespace, c)
+	if err != nil {
+		return err
+	}
+	if err := updateCmap(cmap, data); err != nil {
+		return err
+	}
+	delete(cmap.Data, actionLockField)
+	return configmaps.Deploy(ctx, cmap, cmap.Namespace, c)
+}
+
+//TranslateAppName translate the name into the pipeline name
+func TranslateAppName(name string) string {
```

Comment:
`App` is not a concept in okteto:
```suggestion
func TranslateDeployName(name string) string {
```

Reply:
This have to be used by destroy command and it would be the same implementation

Reply:
Let's avoid the concept `App`.  We are using `Pipeline` or `GitDeploy` in other places